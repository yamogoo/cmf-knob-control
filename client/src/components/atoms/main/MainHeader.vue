<template lang="pug">
header(
    :class="[`${MAIN_APP_PREFIX}-main-header`]"
)
    MainNavigation
</template>

<script setup lang="ts">
import { MAIN_APP_PREFIX } from '@/config';

import MainNavigation from '@/components/atoms/main/MainNavigation.vue'
</script>

<style lang="scss">
$main-header: (
    sm: (size: (height: 48px)),
    md: (size: (height: 60px)),
    lg: (size: (height: 60px))
);

.#{$MAIN_APP_PREFIX}-main-header {
    position: relative;
    display: block;
    // top: 0;
    width: 100%;
    background-color: none;
    border-bottom: 1px solid gray;
    overflow: hidden;

    @include respond-above(lg) {
        min-height: get('lg.size.height', $main-header);
    }

    @include respond-between(md, lg) {
        min-height: get('md.size.height', $main-header);
    }

    @include respond-below(md) {
        min-height: get('sm.size.height', $main-header);
    }
}
</style>