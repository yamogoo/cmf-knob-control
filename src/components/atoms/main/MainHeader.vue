<template lang="pug">
header(
    :class="[`${MAIN_APP_PREFIX}-main-header`]"
)
    MainNavigation
</template>

<script setup lang="ts">
import { MAIN_APP_PREFIX } from '@/config';

import MainNavigation from '@/components/atoms/main/MainNavigation.vue'
</script>

<style lang="scss">
$main-header: (
    sm: (size: (height: 48px)),
    md: (size: (height: 60px)),
    lg: (size: (height: 60px))
);

.#{$MAIN_APP_PREFIX}-main-header {
    position: relative;
    display: block;
    width: 100%;
    background-color: none;
    overflow: hidden;

    @include respond-above(lg) {
        min-height: get('lg.size.height', $main-header);
    }

    @include respond-between(md, lg) {
        min-height: get('md.size.height', $main-header);
    }

    @include respond-below(md) {
        min-height: get('sm.size.height', $main-header);
    }
}
</style>