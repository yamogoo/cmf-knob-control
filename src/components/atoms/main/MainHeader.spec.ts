import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import MainHeader from './MainHeader.vue'

describe('MainHeader', () => {
  test('header should render MainNavigation component', () => {
    const component = mount(MainHeader)

    const navEl = component.find('[data-test-id="main-navigation"]')
    const isNavElExist = navEl.exists()

    expect(isNavElExist).toBe(true)
    expect(isNavElExist).toMatchSnapshot()
  })
})
