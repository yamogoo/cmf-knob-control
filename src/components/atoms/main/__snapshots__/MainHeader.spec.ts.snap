// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainHeader > header should render MainNavigation component 1`] = `true`;
